container.go

```diff
@@ -152,6 +152,34 @@ func (c *Container) SetPid(pid int) error {
 	return c.storeProcess()
 }
 
+func (c *Container) setStateBlockIndex(index int) error {
+	tState := c.state
+	tState.BlockIndex = index
```

Comment:
I am just updating a temporary object here first. That way if the write to filesystem fails, the object in memory and filesystem are in sync